### ISSUE TITLE
Scope resolution operator in array key

### ISSUE THREAD

**Issue by lorderb** [squizlabs/PHP_CodeSniffer#2170]
```php
<?php

const CONSTANT1 = 'key1';

class SniffTest
{
    const CONSTANT2 = 'key2';

    private $values = [
        CONSTANT1 => 'value1', // <- working
        self::CONSTANT2 => 'value2',  // <- Array double arrow not aligned correctly; expected -10 space(s) but found 1
    ];
}
```

`::` seems to cause a problem

**Comment by gsherwood** [comment]
I'll need more help trying to replicate this.

I assume you're using the `Squiz.Arrays.ArrayDeclaration` sniff as that is where the error message is coming from. If I run just that sniff over your sample code, I get these errors:
```
---------------------------------------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 3 LINES
---------------------------------------------------------------------------------------------------------------
 10 | ERROR | [x] Array key not aligned correctly; expected 23 spaces but found 8
    |       |     (Squiz.Arrays.ArrayDeclaration.KeyNotAligned)
 10 | ERROR | [x] Array double arrow not aligned correctly; expected 7 space(s) but found 1
    |       |     (Squiz.Arrays.ArrayDeclaration.DoubleArrowNotAligned)
 11 | ERROR | [x] Array key not aligned correctly; expected 23 spaces but found 8
    |       |     (Squiz.Arrays.ArrayDeclaration.KeyNotAligned)
 12 | ERROR | [x] Closing parenthesis not aligned correctly; expected 22 space(s) but found 4
    |       |     (Squiz.Arrays.ArrayDeclaration.CloseBraceNotAligned)
---------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------------
```
If I asked for a diff report, it suggests correcting the code like this:
```diff
--- temp.php
+++ PHP_CodeSniffer
@@ -7,7 +7,7 @@
     const CONSTANT2 = 'key2';

     private $values = [
-        CONSTANT1 => 'value1',
-        self::CONSTANT2 => 'value2',
-    ];
+                       CONSTANT1       => 'value1',
+                       self::CONSTANT2 => 'value2',
+                      ];
 }
```
What coding standard are you using? What PHPCS version? Any more info to help me replicate?

**Comment by lorderb** [comment]
Nevermind, we are using a custom sniff.
